tests/ErrorTest.php

```diff
@@ -30,4 +30,39 @@ public function it_returns_api_problems() : void
 
         $this->assertXpathMatch('/problem:problem/problem:title', $dom, ['problem' => 'urn:ietf:rfc:7807']);
     }
+
+    /**
+     * @test
+     * @dataProvider languageProvider
+     * @backupGlobals true
+     */
+    public function it_negotiates_locale(string $acceptLanguage, string $expected) : void
+    {
+        $_SERVER['DEFAULT_LOCALE'] = 'de';
+        $_SERVER['POSSIBLE_LOCALES'] = 'en|es';
+
+        self::bootKernel();
+
+        $request = Request::create('/foo');
+        $request->headers->set('Accept-Language', $acceptLanguage);
+
+        $response = self::$kernel->handle($request);
+
+        $this->assertSame(Response::HTTP_NOT_FOUND, $response->getStatusCode());
+        $this->assertSame('application/problem+xml; charset=utf-8', $response->headers->get('Content-Type'));
+        $this->assertSame($expected, $response->headers->get('Content-Language'));
+
+        $dom = FluentDOM::load($response->getContent());
+
+        $this->assertXpathMatch("/problem:problem[@xml:lang='${expected}']", $dom, ['problem' => 'urn:ietf:rfc:7807']);
+    }
+
+    public function languageProvider() : iterable
+    {
+        yield 'Any language' => ['*', 'en'];
```

Comment:
weird that the default is ignored here, though it doesn't intersect the possible locales. Is it a real use case or is it for testing?

Reply:
Purely for testing, just to separate the two cases (`*` negotiates successfully, `fr` doesn't).